keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+            Objects.requireNonNull(credentials);
+            this.credentials = credentials;
+            return this;
+        }
+
+        /**
+         * Sets the logging level for HTTP requests and responses.
+         *
+         * @param logLevel The amount of logging output when sending and receiving HTTP requests/responses.
+         * @return The updated Builder object.
+         */
+        public Builder httpLogDetailLevel(HttpLogDetailLevel logLevel) {
+            httpLogDetailLevel = logLevel;
+            return this;
+        }
+
+        /**
+         * Adds a policy to the set of existing policies that are executed after
+         * {@link com.azure.keyvault.SecretClient} required policies.
+         *
+         * @param policy The retry policy for service requests.
+         * @return The updated Builder object.
+         * @throws NullPointerException if {@code policy} is {@code null}.
+         */
+        public Builder addPolicy(HttpPipelinePolicy policy) {
+            Objects.requireNonNull(policy);
+            policies.add(policy);
+            return this;
+        }
+
+        /**
+         * Sets the HTTP client to use for sending and receiving requests to and from the service.
+         *
+         * @param client The HTTP client to use for requests.
+         * @return The updated Builder object.
+         * @throws NullPointerException if {@code client} is {@code null}.
+         */
+        public Builder httpClient(HttpClient client) {
+            this.httpClient = client;
+            return this;
+        }
+    }
+
+    public static List<HttpPipelinePolicy> getDefaultPolicies(ServiceClientCredentials credentials, String vaultUrl) {
+        //final ApplicationConfigCredentials credentials = new ApplicationConfigCredentials(connectionString);
+        // Closest to API goes first, closest to wire goes last.
+        final List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+        policies.add(new UserAgentPolicy(String.format("Azure-SDK-For-Java/%s (%s)", SDK_NAME, SDK_VERSION)));
+        policies.add(new RequestIdPolicy());
+        policies.add(new RetryPolicy());
+        policies.add(new KeyvaultCredentialsPolicy(credentials, vaultUrl));
+
+        return policies;
+    }
+
+    /**
+     * Sets a secret in a the key vault.
+     * The SET operation adds a secret to the Azure Key Vault.
+     * If the named secret already exists, Azure Key Vault creates a new version of that secret.
+     * This operation requires the secrets/set permission.
+     *
+     * <p>
+     * The name and value fields in {@code secret} are required.
+     * The expires, contentType and and notBefore values value for the Secret are optional.
+     * If not specified, no values are set for the fields.
+     * </p>
+     *
+     * @param secret The Secret object containing information about the secret and its properties.
+     * @throws NullPointerException if {@code secret} is {@code null}.
```

Comment:
Might want to clarify that it is secret, secret.name, and secret.value that must be non-null.

Reply:
updated

Reply:
Added the info in param description too. It is also mentioned in the javadoc.